WeakAuras/RegionTypes/Text.lua

```diff
@@ -179,43 +179,46 @@ local function modify(parent, region, data)
     containsCustomText = true
   end
 
-  if not containsCustomText and type(data.conditions) == "table" then
+  local allFormatters = {}
+  local getter = function(key, default)
+    local fullKey = "displayText_format_" .. key
+    if (data[fullKey] == nil) then
+      data[fullKey] = default
+    end
+    return data[fullKey]
+  end
+
+  allFormatters[data.displayText] = Private.CreateFormatters(data.displayText, getter)
+
+  if type(data.conditions) == "table" then
     for _, condition in ipairs(data.conditions) do
       if type(condition.changes) == "table" then
         for _, change in ipairs(condition.changes) do
           if type(change.property) == "string"
           and change.property == "displayText"
           and type(change.value) == "string"
-          and Private.ContainsCustomPlaceHolder(change.value)
           then
-            containsCustomText = true
-            break
+            if not containsCustomText and Private.ContainsCustomPlaceHolder(change.value) then
+              containsCustomText = true
+            end
+            allFormatters[change.value] = Private.CreateFormatters(change.value, getter)
```

Comment:
I don't think you should be keeping separate list of formatters, that is if two conditions both change the text to %foo, I would expect just one %foo-formatter, and one %foo option. 

So you probably need to extend CrateFormatters to take multiple strings as input. (Or inline it).